src/utils/hooks/useUpdateApplication.tsx

```diff
@@ -0,0 +1,78 @@
+import { ApolloError } from '@apollo/client'
+import { useEffect, useState } from 'react'
+import {
+  ApplicationResponse,
+  Trigger,
+  useGetResponsesInApplicationQuery,
+  useUpdateApplicationMutation,
+} from '../generated/graphql'
+
+interface useUpdateApplicationProps {
+  applicationSerial: string
+  applicationTrigger?: Trigger
+}
+
+const useUpdateApplication = ({ applicationSerial }: useUpdateApplicationProps) => {
+  const [submitted, setSubmitted] = useState(false)
+  const [processing, setProcessing] = useState(false)
+  const [error, setError] = useState<ApolloError | undefined>()
+
+  // Hook to get existing responses in cache - triggered when user submits application
+  const { data, error: responsesError } = useGetResponsesInApplicationQuery({
```

Comment:
Related to the earlier comment -- all the response values have already been loaded by `useGetResponsesAndElementState`. Can you not just pass them into the `useUpdateApplication` hook rather than re-query for them?

Reply:
Like chatted in person - What Apollo client does is that - if the elements are already in cache this query will only get from the cache, so no need to pass every element down the track.